src/fe/fe_lagrange.C

```diff
@@ -831,134 +832,149 @@ void lagrange_compute_constraints (DofConstraints & constraints,
+                        (std::abs(their_dof_value) < .999))
+                      {
+                        constraint_row->emplace(their_dof_g, their_dof_value);
+                      }
+  #ifdef DEBUG
+                    // Protect for the case u_i = 0.999 u_j,
+                    // in which case i better equal j.
+                    else if (their_dof_value >= .999)
+                      libmesh_assert_equal_to (my_dof_g, their_dof_g);
+  #endif
+                  }
               }
+          }
 
-              // The support point of the DOF
-              const Point & support_point = my_side->point(my_dof);
-
-              // Figure out where my node lies on their reference element.
-              const Point mapped_point = FEMap::inverse_map(Dim-1,
-                                                            parent_side.get(),
-                                                            support_point);
-
-              // Compute the parent's side shape function values.
-              for (unsigned int their_dof=0;
-                   their_dof != n_parent_side_dofs; their_dof++)
-                {
-                  libmesh_assert_less (their_dof, parent_side->n_nodes());
-
-                  // Their global dof index.
-                  const dof_id_type their_dof_g =
-                    parent_dof_indices[their_dof];
-
-                  const Real their_dof_value = FEInterface::shape(Dim-1,
-                                                                  side_fe_type,
-                                                                  parent_side.get(),
-                                                                  their_dof,
-                                                                  mapped_point);
-
-                  // Only add non-zero and non-identity values
-                  // for Lagrange basis functions.
-                  if ((std::abs(their_dof_value) > 1.e-5) &&
-                      (std::abs(their_dof_value) < .999))
-                    {
-                      constraint_row->emplace(their_dof_g, their_dof_value);
-                    }
-#ifdef DEBUG
-                  // Protect for the case u_i = 0.999 u_j,
-                  // in which case i better equal j.
-                  else if (their_dof_value >= .999)
-                    libmesh_assert_equal_to (my_dof_g, their_dof_g);
-#endif
-                }
-            }
-        }
+        if (add_p_level)
+          {
+            // p refinement constraints:
+            // constrain dofs shared between
+            // active elements and neighbors with
+            // lower polynomial degrees
+            const unsigned int min_p_level =
+              neigh->min_p_level_by_neighbor(elem, elem->p_level());
+            if (min_p_level < elem->p_level())
+              libmesh_error_msg(
+                "Mismatched p-refinement levels for LAGRANGE is not currently supported");
```

Comment:
We often use mismatched p-level in 1D, since there you don't have a continuity problem. Can we have this new error message allow a mismatch if both neighbors are 1D?

Reply:
This method returns at the top if `Dim == 1`

Reply:
Beautiful early `return` statement